src/server/game/Entities/Player/Player.cpp

```diff
@@ -15852,6 +15858,17 @@ void Player::RemoveRewardedQuest(uint32 questId, bool update /*= true*/)
         m_RewardedQuestsSave[questId] = QUEST_FORCE_DELETE_SAVE_TYPE;
     }
 
+    // Remove seasonal quest also
+    if (Quest const* qInfo = sObjectMgr->GetQuestTemplate(questId))
+    {
+        uint16 eventId = sGameEventMgr->GetEventIdForQuest(qInfo);
```

Comment:
Sorry, my bad. I looked up the function in the `master` branch here on TC and thought I was looking at 3.3.5 source. It is not in the 3.3.5 branch, so you will need to add it in this PR to make it work,